xwiki-platform-core/xwiki-platform-refactoring/xwiki-platform-refactoring-api/src/main/java/org/xwiki/refactoring/job/UserOrientedRequest.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.refactoring.job;
+
+import org.xwiki.job.Request;
+import org.xwiki.model.reference.DocumentReference;
+import org.xwiki.stability.Unstable;
+
+/**
+ * Define a request that can be processed by a user.
+ *
+ * @version $Id$
+ * @since 10.10RC1
+ */
+@Unstable
+public interface UserOrientedRequest extends Request
```

Comment:
Does this really need to be public ? I would keep that as an internal refactoring interface.

If a public interface it's really not in the right place since there is nothing related to refactoring in it.

Reply:
I cannot put it as package protected as I need it in RefactoringScriptService, so yeah I think it's really a public one. 

> If a public interface it's really not in the right place since there is nothing related to refactoring in it.

Any suggestion where I should put it then?

Reply:
Sorry I was not very clear: I was not talking about the `public` keywork but the fact that this should be in an `*.internal.*` package IMO.

Reply:
Oh, right. Then I can refactor to put it in `internal/job`.

Reply:
Hmm actually you seems to expose it in other non internal classes so what I suggested is wrong. So I guess the best would be to move it in a more generic place so that it can be reused by other modules (for example in EM jobs requests since it's the exact same need). Could be a new xwiki-platform-job-security or a less specific xwiki-platform-job-api (I prefer this one I think) for example.

Reply:
So do you suggest I only move my new interface in this new module, or I also move some stuff related to the requests?

Reply:
Any generic tool/api related to job and security checking.

Reply:
About this refactoring, I'd need to keep the same packages for backward compatibility reason, especially for the classes that are not inside "internal" package, shouldn't I?

Reply:
I'm not really asking any refactoring, just putting in the right place the new generic public APIs you add.